server/service/src/sync/translations/purchase_order_line.rs

```diff
@@ -0,0 +1,239 @@
+use chrono::NaiveDate;
+use repository::{
+    ChangelogRow, ChangelogTableName, PurchaseOrderLineRow, PurchaseOrderLineRowRepository,
+    StorageConnection, SyncBufferRow,
+};
+use serde::{Deserialize, Serialize};
+use util::sync_serde::{date_option_to_isostring, empty_str_as_option, zero_date_as_option};
+
+use crate::sync::translations::{
+    master_list::MasterListTranslation, name::NameTranslation, period::PeriodTranslation,
+    purchase_order::PurchaseOrderTranslation, store::StoreTranslation, PullTranslateResult,
+    PushTranslateResult, SyncTranslation,
+};
+
+#[allow(non_snake_case)]
+#[derive(Deserialize, Serialize, Clone, Debug)]
+pub struct LegacyPurchaseOrderLineRow {
+    #[serde(rename = "ID")]
+    pub id: String,
+    #[serde(rename = "purchase_order_ID")]
+    pub purchase_order_id: String,
+    #[serde(default)]
+    pub line_number: Option<i64>,
+    #[serde(default)]
+    #[serde(deserialize_with = "empty_str_as_option")]
+    #[serde(rename = "item_ID")]
+    pub item_link_id: Option<String>,
+    #[serde(default)]
+    #[serde(deserialize_with = "empty_str_as_option")]
+    pub item_name: Option<String>,
+    #[serde(default)]
+    #[serde(rename = "snapshotQuantity")]
+    pub number_of_packs: Option<f64>,
```

Comment:
Hi just diving into this lane. Am currently writing the update mutation for this. And was roaming around OG and found that `snapshotQuantity` is used to render Stock on hand for an item within a purchase order line? Are we going to use it the same way? If so I was just thinking that `number_of_packs` might be confusing because there's another 'number of packs' text used differently. 

I could also be misunderstanding how this works so please let me know :D 

<img width="1266" height="555" alt="Screenshot 2025-07-21 at 12 41 40 PM" src="https://github.com/user-attachments/assets/bcab707e-5576-42c1-99a5-b1b960071e22" />

Reply:
Good catch John, I think number of packs should be renamed to `snapshot_soh` or something similar. `number_of_packs` is confusing to me!

`requested_quantity` I think is too vague to me too, I think we should have either `requested_number_of_units` or `requested_number_of_packs` to make it very clear.